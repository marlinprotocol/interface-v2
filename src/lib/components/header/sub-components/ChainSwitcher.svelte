<script lang="ts">
	import { buttonClasses } from '$lib/atoms/componentClasses';
	import { switchChain } from '$lib/utils/helpers/networkHelper';
	import { environment } from '$lib/data-stores/environment';
	import { staticImages } from '$lib/components/images/staticImages';
	import { connected, web3WalletStore } from '$lib/data-stores/walletProviderStore';
	import {
		allowedChainsStore,
		chainStore,
		updateChainStore
	} from '$lib/data-stores/chainProviderStore';
	import Icon from '$lib/atoms/icons/Icon.svelte';
	import chevronDown from 'svelte-awesome/icons/chevronDown';

	export let id = 'chain-dropdown';

	let activeChainImage = staticImages.chainLogo;
	$: dropdownOpen = false;

	function handleChainSwitch(chainId: number) {
		if ($connected && chainId !== $chainStore.chainId) {
			switchChain(chainId, $web3WalletStore[0].provider);
		} else {
			updateChainStore(chainId);
		}
		dropdownOpen = false; // Close dropdown after switching
	}

	function toggleDropdown() {
		dropdownOpen = !dropdownOpen;
	}

	function closeDropdown() {
		dropdownOpen = false;
	}

	function closeSwitcherWhenClickedOutside(event: MouseEvent) {
		const dropdown = document.getElementById(id);
		if (!dropdown?.contains(event.target as Node)) {
			closeDropdown();
		}
	}

	$: {
		const activeChainId = $chainStore.chainId;
		if (activeChainId !== null) {
			activeChainImage = environment.valid_chains[activeChainId]?.chain_image;
		}
	}
</script>

<svelte:window on:click={(e) => closeSwitcherWhenClickedOutside(e)} />

<div class="dropdown dropdown-end" {id}>
	<!-- svelte-ignore a11y-click-events-have-key-events -->
	<div
		tabindex={$allowedChainsStore.length === 0 ? -1 : 0}
		role="button"
		class="{buttonClasses.whiteFilled} chain-btn m-1 flex h-12"
		on:click={toggleDropdown}
	>
		<div class="flex items-center justify-center">
			<div class="flex h-8 w-8 items-center justify-center">
				{#if activeChainImage}
					<img src={activeChainImage} alt="current chain" />
				{:else}
					<span class="flex h-8 w-8 items-center justify-center rounded-full bg-primary text-white">
						{$chainStore.chainDisplayName.charAt(0).toLocaleUpperCase()}
					</span>
				{/if}
			</div>
			<div class="ml-2">
				<Icon data={chevronDown} size={10} iconColorClass="icon-primary" />
			</div>
		</div>
	</div>
	{#if dropdownOpen}
		<!-- svelte-ignore a11y-no-noninteractive-tabindex -->
		<ul
			tabindex={$allowedChainsStore.length === 0 ? -1 : 0}
			class="menu dropdown-content rounded-box z-[10] mt-4 w-52 bg-base-100 p-2 shadow"
		>
			{#each $allowedChainsStore as chain (chain)}
				<li class="flex {$chainStore.chainId === chain ? 'rounded-lg bg-primary text-white' : ''}">
					<button on:click={() => handleChainSwitch(chain)}>
						<div class="h-6 w-6 rounded-full ring-1 ring-white">
							<img src={environment.valid_chains[chain].chain_image} alt="Chain Logo" />
						</div>
						{environment.valid_chains[chain].chain_name}
					</button>
				</li>
			{/each}
		</ul>
	{/if}
</div>
