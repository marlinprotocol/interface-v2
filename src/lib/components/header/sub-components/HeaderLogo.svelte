<a href="/">
	<img class="hidden sm:block" src="/logo/logo-name.svg" alt="Marlin Logo" width="140" />
	<img class="block sm:hidden" src="/logo/marlin-logo.svg" alt="Marlin Logo" width="40" />
</a>
