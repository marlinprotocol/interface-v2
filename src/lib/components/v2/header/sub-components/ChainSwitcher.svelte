<script lang="ts">
	import { buttonClasses } from '$lib/atoms/componentClasses';
	import { switchChain } from '$lib/utils/helpers/networkHelper';
	import { environment } from '$lib/data-stores/environment';
	import { staticImages } from '$lib/components/images/staticImages';
	import { connected, web3WalletStore } from '$lib/data-stores/walletProviderStore';
	import {
		allowedChainsStore,
		chainStore,
		updateChainStore
	} from '$lib/data-stores/chainProviderStore';
	import Icon from '$lib/atoms/icons/Icon.svelte';
	import chevronDown from 'svelte-awesome/icons/chevronDown';

	let activeChainImage = staticImages.chainLogo;

	function handleChainSwitch(chainId: number) {
		if ($connected && chainId !== $chainStore.chainId) {
			switchChain(chainId, $web3WalletStore[0].provider);
		} else {
			updateChainStore(chainId);
		}
	}

	function closeSwitcherWhenClickedOutside(event: MouseEvent) {
		const dropdown = document.getElementById('chain-dropdown');
		if (!dropdown?.contains(event.target as Node)) {
			dropdown?.removeAttribute('open');
		}
	}

	$: {
		const activeChainId = $chainStore.chainId;
		if (activeChainId !== null) {
			activeChainImage = environment.valid_chains[activeChainId]?.chain_image;
		}
	}
</script>

<svelte:window on:click={(e) => closeSwitcherWhenClickedOutside(e)} />

<div class="dropdown dropdown-end" id="chain-dropdown">
	<div
		tabindex={$allowedChainsStore.length === 0 ? -1 : 0}
		role="button"
		class="{buttonClasses.whiteFilled} chain-btn m-1 flex h-12"
	>
		<div class="flex items-center justify-center">
			<div class="h-8 w-8">
				<img src={activeChainImage} alt="current chain" />
			</div>
			<div class="ml-2">
				<Icon data={chevronDown} size={10} iconColorClass="icon-primary" />
			</div>
		</div>
	</div>
	<!-- svelte-ignore a11y-no-noninteractive-tabindex -->
	<ul
		tabindex={$allowedChainsStore.length === 0 ? -1 : 0}
		class="menu dropdown-content rounded-box z-[1] mt-4 w-52 bg-base-100 p-2 shadow"
	>
		{#each $allowedChainsStore as chain (chain)}
			<li class="flex {$chainStore.chainId === chain ? 'rounded-lg bg-primary text-white' : ''}">
				<button on:click={() => handleChainSwitch(chain)}>
					<div class="h-6 w-6 rounded-full ring-1 ring-white">
						<img src={environment.valid_chains[chain].chain_image} alt="Chain Logo" />
					</div>
					{environment.valid_chains[chain].chain_name}
				</button>
			</li>
		{/each}
	</ul>
</div>
