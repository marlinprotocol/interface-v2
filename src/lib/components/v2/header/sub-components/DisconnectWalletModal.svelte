<script lang="ts">
	import Button from '$lib/atoms/v2/buttons/Button.svelte';
	import Modal from '$lib/atoms/v2/modals/Modal.svelte';
	import TooltipIcon from '$lib/atoms/v2/tooltips/TooltipIcon.svelte';
	import { staticImages } from '$lib/components/images/staticImages';
	import AddTokenPrompt from '$lib/components/v2/prompts/AddTokenPrompt.svelte';
	import { chainConfigStore } from '$lib/data-stores/chainProviderStore';
	import { walletStore } from '$lib/data-stores/walletProviderStore';
	import { handleCopyClick } from '$lib/utils/v2/helpers/componentHelper';

	export let modalFor: string;
	export let disconnect: () => void;
</script>

<Modal {modalFor}>
	<svelte:fragment slot="title">Your wallet</svelte:fragment>
	<!-- <svelte:fragment slot="successmsg">Conversion Successful</svelte:fragment> -->
	<svelte:fragment slot="content">
		<div class="mb-4 rounded-lg bg-base-200 p-4 text-left">
			<div class="flex items-center text-sm font-normal">
				My Address
				<TooltipIcon
					tooltipDirection="tooltip-right"
					tooltipText="Connected wallet address"
					styleClass="ml-1"
				/>
			</div>
			<div class="flex items-center">
				<div
					data-testid="wallet-address"
					class="mt-1 w-[98%] overflow-hidden overflow-ellipsis text-xl font-semibold"
				>
					{$walletStore.address}
				</div>
				<button
					class="shrink-0 cursor-pointer"
					on:click={() => handleCopyClick($walletStore.address, 'Address copied to clipboard')}
				>
					<img width={18} src={staticImages.copyIcon} alt="Copy" />
				</button>
			</div>
		</div>
		<div class="flex flex-wrap gap-4">
			<AddTokenPrompt tokenFor="POND" label="POND" />
			<AddTokenPrompt tokenFor="MPOND" label="MPond" />
			<a
				href="{$chainConfigStore.block_explorer_url}/address/{$walletStore.address}"
				target="_blank"
				rel="noopener noreferrer"
			>
				<Button variant="greyFilled" size="small" styleClass="font-medium gap-3  flex-nowrap">
					<img src={staticImages.OpenInNew} alt="Open" />
					{$chainConfigStore.block_explorer_name}
				</Button>
			</a>
		</div>
	</svelte:fragment>
	<svelte:fragment slot="actionButtons">
		<Button
			variant="filled"
			size="large"
			onclick={disconnect}
			styleClass="w-full font-semibold text-base"
		>
			LOGOUT
		</Button>
	</svelte:fragment>

	<!-- <svelte:fragment slot="errorMessage">
		Insufficient MPOND
	</svelte:fragment> -->
</Modal>
