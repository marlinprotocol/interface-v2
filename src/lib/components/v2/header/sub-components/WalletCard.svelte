<script lang="ts">
	import Text from '$lib/atoms/v2/texts/Text.svelte';

	export let imageSrc: string;
	export let title: string;
</script>

<div class="flex w-1/2 flex-col items-center overflow-hidden rounded-lg bg-base-200 p-4">
	{#if imageSrc}
		<img src={imageSrc} alt={title} width="64px" />
	{/if}
	<Text text={title} variant="h4" styleClass="mt-4 mb-6 font-semibold font-sans" />
	<slot name="button" />
</div>
