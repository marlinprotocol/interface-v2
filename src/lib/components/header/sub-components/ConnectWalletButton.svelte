<script lang="ts">
	import Button from '$lib/atoms/buttons/Button.svelte';
	import lock from 'svelte-awesome/icons/lock';
	import Icon from '$lib/atoms/icons/Icon.svelte';
	import onboard from '$lib/controllers/web3OnboardController';
	import ChainSwitcher from '$lib/components/header/sub-components/ChainSwitcher.svelte';
	import { cn } from '$lib/utils/helpers/commonHelper';

	export let styleClass = '';
	export let isLarge = false;
	export let connectButtonText = 'Connect Wallet';

	const connect = async () => {
		console.log('connecting to the wallet...');
		const connection = await onboard.connectWallet();
		console.log('connection', connection);
	};
</script>

{#if isLarge}
	<Button
		onclick={connect}
		styleClass={cn('flex gap-[10.3px] w-full h-14 text-base font-semibold', styleClass)}
	>
		<Icon data={lock} size={20} iconColorClass="icon-white" />
		Connect Wallet
	</Button>
{:else}
	<div data-testid="connect-wallet-button" class={cn('flex items-center gap-2', styleClass)}>
		<ChainSwitcher />
		<Button
			onclick={connect}
			size="small"
			variant="filled"
			styleClass="flex gap-[10.3px] w-fit text-sm h-12 flex items-center cnt-btn"
		>
			{connectButtonText}
		</Button>
	</div>
{/if}
