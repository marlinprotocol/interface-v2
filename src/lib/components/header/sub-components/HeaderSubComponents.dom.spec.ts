import { render } from '@testing-library/svelte';
import WalletCard from './WalletCard.svelte';
import HeaderLogo from './HeaderLogo.svelte';
import ConnectWalletButton from './ConnectWalletButton.svelte';
import { describe, it, expect } from 'vitest';
import html from '@playpilot/svelte-htm';
import ChainSwitcher from './ChainSwitcher.svelte';
import DisconnectWalletModal from './DisconnectWalletModal.svelte';

describe('ChainSwitcher', () => {
	it('renders the ChainSwitcher component properly', () => {
		const { container } = render(ChainSwitcher);
		expect(container).toMatchSnapshot();
	});
});

describe('ConnectWalletButton', () => {
	it('renders the ConnectWalletButton component properly', () => {
		const { container } = render(HeaderLogo);
		expect(container).toMatchSnapshot();
	});

	it('renders large button layout when isLarge is true', () => {
		const { getByText } = render(ConnectWalletButton, {
			isLarge: true
		});

		expect(getByText('Connect Wallet')).toBeTruthy();
	});

	it('renders small button layout with chain switcher and custom text when isLarge is false', () => {
		const { getByText } = render(ConnectWalletButton, {
			isLarge: false,
			connectButtonText: 'Custom Connect'
		});
		expect(getByText('Custom Connect')).toBeTruthy();
	});
});

describe('Disconnect Wallet Modal', () => {
	it('renders the Disconnect Wallet Modal component properly', () => {
		const { container } = render(DisconnectWalletModal);
		expect(container).toMatchSnapshot();
	});
});

describe('HeaderLogo', () => {
	it('renders the HeaderLogo component properly', () => {
		const { container } = render(HeaderLogo);
		expect(container).toMatchSnapshot();
	});
	it('renders both images with correct sources and one is initially hidden', () => {
		const { getAllByAltText } = render(HeaderLogo);

		const images = getAllByAltText('Marlin Logo');

		// Expect two images to be found
		expect(images.length).toBe(2);

		// Check if the right logos have the right src
		const largeScreenLogo = images.find((img) => img.getAttribute('class')?.includes('sm:block'));
		const smallScreenLogo = images.find((img) => img.getAttribute('class')?.includes('sm:hidden'));

		// Since we cannot test screen sizes here, we'll just check if the src is correct
		expect(largeScreenLogo?.getAttribute('src')).toBe('/logo/logo-name.svg');
		expect(smallScreenLogo?.getAttribute('src')).toBe('/logo/marlin-logo.svg');
	});
});

describe('WalletCard', () => {
	it('renders the WalletCard component properly', () => {
		const { container } = render(WalletCard, {
			props: { imageSrc: 'example.png', title: 'small card' }
		});
		expect(container).toMatchSnapshot();
	});

	it('renders the title text', () => {
		const titleText = 'Sample Title';

		const { getByText } = render(WalletCard, {
			props: {
				title: titleText,
				imageSrc: '' // Pass an empty string because we are testing the title here
			}
		});
		expect(getByText(titleText)).toBeTruthy();
	});

	it('renders an image when imageSrc is provided', () => {
		const testImageUrl = 'test-image.jpg';
		const testTitle = 'Test Image';

		const { getByAltText } = render(WalletCard, {
			props: {
				imageSrc: testImageUrl,
				title: testTitle
			}
		});

		const image = getByAltText(testTitle);
		expect(image).toBeTruthy();
		expect(image.getAttribute('src')).toBe(testImageUrl);
	});

	it('does not render an image when imageSrc is not provided', () => {
		const { container } = render(WalletCard, {
			props: {
				title: 'No Image Title',
				imageSrc: ''
			}
		});

		// Here we are checking that no <img> elements are present in the container.
		expect(container.querySelector('img')).toBeNull();
	});

	it('allows content to be added via slots', async () => {
		const slotContentText = 'Click me';
		const { getByText } = render(html`
                <${WalletCard} title=${'Card with Button'} imageSrc='example.png'>
                 <div slot='button'><button>${slotContentText}</button></div>
                </${WalletCard}>
              `);
		expect(getByText(slotContentText)).toBeTruthy();
	});
});
