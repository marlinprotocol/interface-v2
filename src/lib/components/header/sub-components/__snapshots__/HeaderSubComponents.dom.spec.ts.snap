// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChainSwitcher > renders the ChainSwitcher component properly 1`] = `
<body>
  <onboard-v2
    style="all: initial;"
  />
  <style>
    
    
@font-face {
  font-family: 'Inter';
  font-style:  normal;
  font-weight: 300 600;
  font-display: swap;
  src: url("https://rsms.me/inter/font-files/Inter-roman.var.woff2?v=3.19") format("woff2");
}

  
  </style>
  <div>
    <div
      class="dropdown dropdown-end"
      id="chain-dropdown"
    >
      <div
        class="btn btn-theme btn-sm text-primary px-4 border-primary bg-inherit hover:text-primary-focus hover:border-primary-focus hover:bg-inherit disabled:bg-inherit disabled:text-grey-300 disabled:border-grey-300 disabled:cursor-not-allowed chain-btn m-1 flex h-12"
        role="button"
        tabindex="0"
      >
        <div
          class="flex items-center justify-center"
        >
          <div
            class="flex h-8 w-8 items-center justify-center"
          >
            <img
              alt="current chain"
              src="/images/arbitrum.webp"
            />
          </div>
           
          <div
            class="ml-2"
          >
            <svg
              aria-label=""
              class="fa-icon icon-info s-T5YyQewRBd7n"
              data-testid="icon"
              height="16"
              role="presentation"
              style="width:10px; height:10px; cursor:pointer;  "
              version="1.1"
              viewBox="0 0 1792 1792"
              width="16"
            >
              <path
                d="M1683 808l-742 741q-19 19-45 19t-45-19l-742-741q-19-19-19-45.5t19-45.5l166-165q19-19 45-19t45 19l531 531 531-531q19-19 45-19t45 19l166 165q19 19 19 45.5t-19 45.5z"
              />
              
               
               
              
              
            </svg>
            <!--&lt;Svg&gt;-->
            <!--&lt;Icon&gt;-->
            <!--&lt;Icon&gt;-->
          </div>
        </div>
      </div>
       
    </div>
    <!--&lt;ChainSwitcher&gt;-->
  </div>
</body>
`;

exports[`Disconnect Wallet Modal > renders the Disconnect Wallet Modal component properly 1`] = `
<body>
  <onboard-v2
    style="all: initial;"
  />
  <style>
    
    
@font-face {
  font-family: 'Inter';
  font-style:  normal;
  font-weight: 300 600;
  font-display: swap;
  src: url("https://rsms.me/inter/font-files/Inter-roman.var.woff2?v=3.19") format("woff2");
}

  
  </style>
  <div>
    <input
      class="modal-toggle s-GZwqZrPaBy6h"
      id=""
      tabindex="-1"
      type="checkbox"
    />
     
    <div
      class="modal s-GZwqZrPaBy6h"
      data-testid="modal"
    >
      <div
        class="modal-box flex flex-col rounded-3xl bg-white-100 p-0 drop-shadow-sm w-11/12 sm:w-3/4 sm:max-w-[607px] overflow-x-hidden overflow-y-visible s-GZwqZrPaBy6h"
      >
        <div
          class="modal-header sticky top-0 z-10 flex items-center justify-between bg-white-100 px-6 pt-6 s-GZwqZrPaBy6h"
        >
          <div
            class="flex w-full flex-col s-GZwqZrPaBy6h"
          >
             
             
             
            <div
              class="text-left text-[22px] font-medium tracking-[-1px] s-GZwqZrPaBy6h"
            >
              Your wallet
            </div>
             
          </div>
           
          <button
            class="flex h-[54px] w-[54px] shrink-0 cursor-pointer items-center justify-center self-baseline rounded-full border border-grey-100 bg-secondary-content text-sm font-bold text-gray-300 s-GZwqZrPaBy6h"
            data-testid="modal-close-button"
          >
            <img
              alt="Close"
              class="icon-invert s-GZwqZrPaBy6h"
              src="/images/closeIcon.svg"
            />
          </button>
        </div>
         
        <div
          class="modal-body px-6 pt-4 s-GZwqZrPaBy6h"
        >
          <div
            class="mb-4 rounded-lg bg-base-200 p-4 text-left"
          >
            <div
              class="flex items-center text-sm font-normal"
            >
              My Address
				
              <div
                class="tooltip ml-[14px] flex shrink-0 text-left font-normal ml-1 tooltip-secondary tooltip-right"
                data-testid="tooltip-icon"
                data-tip="Connected wallet address"
              >
                <img
                  alt="Info"
                  src="/images/alertV2Icon.svg"
                  width="14px"
                />
              </div>
              <!--&lt;TooltipIcon&gt;-->
            </div>
             
            <div
              class="flex items-center"
            >
              <div
                class="mt-1 w-[98%] overflow-hidden overflow-ellipsis text-xl font-semibold"
                data-testid="wallet-address"
              >
                
              </div>
               
              <button
                class="shrink-0 cursor-pointer"
              >
                <img
                  alt="Copy"
                  src="/images/copyIconNew.svg"
                  width="18"
                />
              </button>
            </div>
          </div>
           
          <div
            class="flex flex-wrap gap-4"
          >
            <button
              class="h-10 btn btn-theme btn-sm items-center flex justify-center disabled:cursor-not-allowed !rounded-[100px] bg-base-200 text-primary border-none hover:bg-base-300 font-medium gap-3"
              id=""
            >
               
              <span
                class="text-2xl font-light"
              >
                +
              </span>
               Add 
              POND
            </button>
            
            <!--&lt;Button&gt;-->
            <!--&lt;AddTokenPrompt&gt;-->
             
            <button
              class="h-10 btn btn-theme btn-sm items-center flex justify-center disabled:cursor-not-allowed !rounded-[100px] bg-base-200 text-primary border-none hover:bg-base-300 font-medium gap-3"
              id=""
            >
               
              <span
                class="text-2xl font-light"
              >
                +
              </span>
               Add 
              MPond
            </button>
            
            <!--&lt;Button&gt;-->
            <!--&lt;AddTokenPrompt&gt;-->
             
            <a
              href="https://sepolia.arbiscan.io/address/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <button
                class="h-10 btn btn-theme btn-sm items-center flex justify-center disabled:cursor-not-allowed !rounded-[100px] bg-base-200 text-primary border-none hover:bg-base-300 font-medium gap-3 flex-nowrap"
                id=""
              >
                 
                <img
                  alt="Open"
                  src="/images/openinnew.svg"
                />
                 
                Arbiscan
              </button>
              
              <!--&lt;Button&gt;-->
            </a>
          </div>
        </div>
         
        <div
          class="modal-footer sticky bottom-0 z-[2] bg-white-100 px-6 pb-6 pt-4 s-GZwqZrPaBy6h"
        >
          <button
            class="gap-1.5 h-14 btn btn-theme btn-sm items-center flex justify-center disabled:cursor-not-allowed !rounded-[100px] btn-primary disabled:bg-primary disabled:bg-opacity-60 disabled:text-white w-full font-semibold text-base"
            id=""
          >
             
            LOGOUT
          </button>
          
          <!--&lt;Button&gt;-->
        </div>
         
      </div>
    </div>
    <!--&lt;Modal&gt;-->
    <!--&lt;DisconnectWalletModal&gt;-->
  </div>
</body>
`;
